src/System.CommandLine/Option.cs

```diff
@@ -12,7 +12,7 @@ namespace System.CommandLine
     /// <summary>
     /// A symbol defining a named parameter and a value for that parameter. 
     /// </summary>
-    public abstract class Option : Symbol, IValueDescriptor
+    public abstract class Option : Symbol
```

Comment:
@adamsitnik, do you have any recommendation what should be used here: https://github.com/dotnet/runtime/blob/f8bd9d29be973e0a9b82b87b9e50a0d3a77d1697/src/coreclr/tools/Common/CommandLineHelpers.cs#L187-L189? It (`--make-repro`) is a special facility where we create a repro package with command+arguments+artifacts (as zip). In there, we are checking if `Option`'s value is default (set by default factory). Note this is not a hot path so using reflection is an option as a last resort. (to get by the compile error for now, I used reinterpretation style cast: `(IValueDescriptor)(object)option` 😅)

ps - I am in the middle of grand upgrade in runtime and jitutils repos. There is no rush, I'm gradually rebasing and picking up latest daily versions as I go further in my branch.

Reply:
Found a solution:

```diff
-                    IValueDescriptor descriptor = option;
                    object val = res.CommandResult.GetValue(option);
-                    if (val is not null && !(descriptor.HasDefaultValue && descriptor.GetDefaultValue().Equals(val)))
+                    if (val is not null && !(option.DefaultValueFactory is not null && option.DefaultValueFactory.Invoke().Equals(val)))
```
It is a bit strange that `CliArgument<T>` provides `HasDefaultValue and `GetDefaultValue()` but `CliOption<T>` does not. 👀

Reply:
https://github.com/dotnet/runtime/pull/84229